File: src/Entity/Browser/Header/Tab.php
<?php

declare(strict_types=1);

namespace Yggverse\Yoda\Entity\Browser\Header;

use \Yggverse\Yoda\Entity\Browser\Header;

class Tab
{
    public \GtkButton $gtk;

    // Dependencies
    public Header $header;

    // Defaults
    protected string $_label = '+';

    public function __construct(
        Header $header
    ) {
        // Init dependency
        $this->header = $header;

        // Init GTK
        $this->gtk = new \GtkButton;

        $this->gtk->set_label(
            $this->_label
        );

        // Render
        $this->gtk->show();

        // Init events
        $this->gtk->connect(
            'clicked',
            function(
                \GtkButton $entity
            ) {
                $this->header->browser->container->tab->append(
                    null,
                    false
                );
            }
        );
    }
}

File: src/app/browser/header/menu/main/tab.cpp
#include "tab.h"

namespace app::browser::header::menu::main
{
    // Construct
    Tab::Tab(
        Main *main
    ) {
        // Init dependencies
        this->main = main;

        // Init model
        this->model = g_menu_new();

        // Init menu
        this->gtk = gtk_popover_menu_new_from_model(
            G_MENU_MODEL(
                this->model
            )
        );
    }
}
